src/Nox.Types/Types/CultureCode/CultureCode.cs

```diff
@@ -15,14 +29,41 @@ internal override ValidationResult Validate()
     {
         var result = base.Validate();
 
-        if (!CultureCodeRegex().IsMatch(Value))
+        if (!SupportedCultureCodes.Contains(Value))
         {
             result.Errors.Add(new ValidationFailure(nameof(Value), $"Could not create a Nox CultureCode type with unsupported value '{Value}'."));
         }
 
         return result;
     }
+    
+    /// <summary>
+    /// Creates a new instance of <see cref="CultureCode"/>
+    /// <param name="value">The Culture value type of <see cref="Culture"/> to create the <see cref="CultureCode"/> with</param>
+    /// </summary>
+    /// <returns>A new instance of <see cref="CultureCode"/></returns>
+    public static CultureCode From(Culture value)
+    {
+       
+        var field =  value.GetType().GetField(value.ToString());
+        DisplayNameAttribute? attribute = null;
+        if(field is not null)
+        {
+            attribute = field.GetCustomAttributes(typeof(DisplayNameAttribute), false).FirstOrDefault() as DisplayNameAttribute;
+        }
+        var cultureCode = attribute?.DisplayName ?? value.ToString();
+        var newObject = new CultureCode
```

Comment:
statement from here to the end of the method  can be replaced by:
```suggestion
       return From(cultureCode);
```

Reply:
Desired changens have been applied

Reply:
@msadikkoseaupa I'm not seeing the change?

`public static CultureCode From(Culture value)
{
   return From(value.ToDisplayName());
}`

Reply:
Sorry for the missunderstanding. Desired changens have been applied